test/pools/disjoint_pool.cpp

```diff
@@ -58,8 +58,13 @@ TEST_F(test, freeErrorPropagation) {
     void *ptr = umfPoolMalloc(pool, size);
 
     freeReturn = UMF_RESULT_ERROR_MEMORY_PROVIDER_SPECIFIC;
+    // this umfPoolFree() will not free the memory
```

Comment:
and why is that...?

// that's me asking, not knowing what's inside UMF repo 😉 you're calling umfPoolFree twice and the second one magically will do the freeing?  :D

Reply:
Yes, take a look 30 lines above:
```c
        umf_result_t free(void *ptr, [[maybe_unused]] size_t size) noexcept {
            if (freeReturn == UMF_RESULT_SUCCESS) {
                ::free(ptr);
            }
            return freeReturn;
        }
```

Reply:
heh, I was looking for some explanation (at best, added to the comment), but if you don't have it for me then no biggie, it's ok.

Reply:
What do you want me to explain more? What needs to be explained?

Reply:
why do we need a second (exactly the same) free call, one after another?

but as I said, if it's just me or you don't have a good explanation for me, then ok, let's move on :)

Reply:
@lukaszstolarczuk This test works in the way that we modify the default behavior of the free() so we can force it to either do the actual free or return the expected error code. When we call this function for the first time we expect it to fail (and we test that) while the second time we do the cleanup to avoid memory leak.

Reply:
And that's the reasonable answer - can we put a comment like this into the description of the test, or near the second free?
Since I was looking for the first time ever at this test, that was certainly not clear to me. Perhaps, if the first free was called e.g. `mock_free` then it would make more sense... just saying 😄

Reply:
I will create a separate small PR for that today